bazelisk.sh

```diff
@@ -31,20 +31,23 @@ declare -A architectures=(
 )
 
 function os_arch() {
-    local arch="$(uname -m -o)"
+    local arch
+    arch="$(uname -m -o)"
     echo "${architectures[$arch]:-${arch}}"
 }
 
 function check_hash() {
-    local file="$1"
-    local target="$(os_arch)"
-    local value="$(sha256sum "${file}" | cut -f1 -d' ')"
+    local file target value
+    file="$1"
+    target="$(os_arch)"
+    value="$(sha256sum "${file}" | cut -f1 -d' ')"
     local expect="${hashes[$target]}"
-    return $(test "$value" == "$expect")
+    test "$value" == "$expect"
```

Comment:
> Separately, did shellcheck want you to pre-declare variables as `local`? If so, did you miss the `expect` variable?

This is [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155) which is concerned that doing something like `local var=$(...)` will mask any errors in the command substitution. In this case, it's just a variable substitution which will not error out.

Reply:
Wow, SC2155 is new to me. I agree with your reasoning :) And I assume shellcheck agrees as well, or else you'd have to write a suppression line.

Reply:
Did you have thoughts on synthesizing the input line for `sha256sum --check` (first comment)? Totally fine if that's out of scope, just thought it would make it a little cleaner.

Reply:
I've added a commit to implement your suggestion.